x-pack/plugins/endpoint/server/routes/alerts/list/lib.ts

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { get } from 'lodash';
+import { encode, decode, RisonValue } from 'rison-node';
+import { SearchResponse } from 'elasticsearch';
+import { KibanaRequest } from 'kibana/server';
+import { Direction } from '../../../../../../legacy/plugins/siem/public/graphql/types';
+import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
+import { Filter, TimeRange, esQuery, esKuery } from '../../../../../../../src/plugins/data/server';
+import {
+  EndpointAppConstants,
+  AlertData,
+  AlertDataWrapper,
+  AlertHits,
+  AlertResultList,
+  ESTotal,
+} from '../../../../common/types';
+import { EndpointAppContext } from '../../../types';
+import {
+  AlertListRequestQuery,
+  AlertListRequestQueryInternal,
+  AlertListRequest,
+  AlertListESRequestBody,
+  AlertListSort,
+} from './types';
+
+export const getRequestData = async (
+  request: KibanaRequest<unknown, AlertListRequestQuery, unknown>,
+  endpointAppContext: EndpointAppContext
+): Promise<AlertListRequestQueryInternal> => {
+  const config = await endpointAppContext.config();
+  const reqData: AlertListRequestQueryInternal = {
+    // Defaults not enforced by schema
+    pageSize: request.query.page_size || config.alertResultListDefaultPageSize,
+    sort: request.query.sort || config.alertResultListDefaultSort,
+    order: request.query.order || (config.alertResultListDefaultOrder as Direction),
+
+    // Filtering
+    query: request.query.query,
+    filters:
+      request.query.filters !== undefined
+        ? ((decode(request.query.filters) as unknown) as Filter[])
+        : ([] as Filter[]),
+    dateRange: (decode(request.query.date_range) as unknown) as TimeRange,
+
+    // Paging
+    pageIndex: request.query.page_index,
+    searchAfter: request.query.after,
+    searchBefore: request.query?.before,
+  };
+
+  if (reqData.searchAfter === undefined && reqData.searchBefore === undefined) {
+    // simple pagination
+    if (reqData.pageIndex === undefined) {
+      reqData.pageIndex = config.alertResultListDefaultFirstPageIndex;
+    }
+    reqData.fromIndex = reqData.pageIndex! * reqData.pageSize;
```

Comment:
Why do we need a `!` here?

Reply:
@peluja1012 `pageIndex` can be undefined, but we know from the statement above that it won't be in this case.